src/go/rdctl/pkg/reg/reg.go

```diff
@@ -0,0 +1,160 @@
+package reg
+
+import (
+	"encoding/json"
+	"fmt"
+	options "github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/options/generated"
+	"reflect"
+	"strings"
+	"unicode/utf16"
+)
+
+func escape(s string) string {
+	s1 := strings.ReplaceAll(s, "\\", "\\\\")
+	return strings.ReplaceAll(s1, `"`, `\\"`)
+}
+
+// reflect the structure into lines for a reg file
+// params:
+// pathParts: represents the registry path to the current item
+// v: the reflected value of the current field
+// jsonTag: the name of the field, used in json (and the registry)
+//
+// returns:
+//
+//	an array of lines representing the current value
+//	boolean: true if the current entry is empty
+//	error
+func convertToRegFormat(pathParts []string, v reflect.Value, jsonTag string) ([]string, bool, error) {
+	kind := v.Kind()
+	switch kind {
+	case reflect.Struct:
+		numFields := v.NumField()
+		scalarReturnedLines := make([]string, 0, numFields)
+		nestedReturnedLines := []string{}
+		for i := 0; i < numFields; i++ {
+			fieldTag := v.Type().Field(i).Tag.Get("json")
+			fieldName := strings.Replace(fieldTag, ",omitempty", "", 1)
```

Comment:
This is less code, since we're working with a structure that we control. Otherwise I'm either getting the position of the command and then a substring before it, or splitting the string on commas and taking the first string.